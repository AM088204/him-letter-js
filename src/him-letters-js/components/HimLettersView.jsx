import React from 'react';
import ContentContainer from 'terra-content-container';
import Spacer from 'terra-spacer';
import { injectIntl } from 'react-intl';
import HimLettersPresenter from './HimLettersPresenter';

//To display DistributeLetters and LetterTemplates tabs
const HimLettersView = () => {
  
  return (
    <ContentContainer>
      <Spacer paddingTop="large+1" paddingBottom="large+1" paddingLeft="large+1" paddingRight="large+1">
        <div>
          <HimLettersPresenter/>
        </div>
      </Spacer>
    </ContentContainer>
  );
};

export default injectIntl(HimLettersView);