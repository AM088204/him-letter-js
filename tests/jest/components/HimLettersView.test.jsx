import React from 'react';
import { mockIntl, shallowWithIntl } from 'terra-enzyme-intl';
import { ApplicationIntlContext } from 'orion-application/lib/application-intl';
import HimLettersView from '../../../src/him-letters-js/components/HimLettersView';

it('should render a HimLettersView with no data', () => {
  jest.spyOn(React, 'useContext').mockImplementation((contextType) => {
    if (contextType === ApplicationIntlContext) {
      return mockIntl;
    }
    return undefined;
  });
  const himLettersView = shallowWithIntl(<HimLettersView />);
  expect(himLettersView.first().shallow()).toMatchSnapshot();
});

it('should render a HimLettersView that has succeeded with a valid HimLettersEngine', () => {
  jest.spyOn(React, 'useContext').mockImplementation((contextType) => {
    if (contextType === ApplicationIntlContext) {
      return mockIntl;
    }
    return undefined;
  });
  const himLettersView = shallowWithIntl(<HimLettersView exampleData="HimLettersJs" />);
  expect(himLettersView.first().shallow()).toMatchSnapshot();
});
