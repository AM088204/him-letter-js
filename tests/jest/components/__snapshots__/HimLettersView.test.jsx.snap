// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a HimLettersView that has succeeded with a valid HimLettersEngine 1`] = `
<ContentContainer
  fill={false}
>
  <Spacer
    isInlineBlock={false}
    marginBottom="none"
    marginLeft="none"
    marginRight="none"
    marginTop="none"
    paddingBottom="large+1"
    paddingLeft="large+1"
    paddingRight="large+1"
    paddingTop="large+1"
  >
    <div>
      <InjectIntl(HimLettersPresenter) />
    </div>
  </Spacer>
</ContentContainer>
`;

exports[`should render a HimLettersView with no data 1`] = `
<ContentContainer
  fill={false}
>
  <Spacer
    isInlineBlock={false}
    marginBottom="none"
    marginLeft="none"
    marginRight="none"
    marginTop="none"
    paddingBottom="large+1"
    paddingLeft="large+1"
    paddingRight="large+1"
    paddingTop="large+1"
  >
    <div>
      <InjectIntl(HimLettersPresenter) />
    </div>
  </Spacer>
</ContentContainer>
`;
